### DISCUSSION TITLE
How to set open / closed shadowroot

### DISCUSSION THREAD

**Discussion by aschaepper** [lit/lit#2428] [Q&A] [answered]
Hello together

According to the documentation here:
https://lit.dev/docs/components/shadow-dom/
and indirectly here: https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow

The static shadowRootOptions field can be used to set the shadowroot mode.
It lists this code:
`
  class DelegatesFocus extends LitElement {
  static shadowRootOptions = {...LitElement.shadowRootOptions, delegatesFocus: true};
}`

My assumption was I could do something like this, because I see in the intellisense from vs code that it accepts a ShadowRootMode enum:
`
  class DelegatesFocus extends LitElement {
    static shadowRootOptions = {...LitElement.shadowRootOptions, mode: ShadowRootMode.open, delegatesFocus: true};
}`

I found this enum in the definition inside vscode in the file "lib.dom.d.ts", but I cant find out which "import" I have to do. I already tried things like this:
`
import { html, css, LitElement, ShadowRootMode} from 'lit';
import { html, css, LitElement, ShadowRoot} from 'lit';
`
Basically this is gessing and I have to be hones I don't know the way of finding this out in typescript myself so this is the reason I have to ask this kind of beginner / rookie question here?

If you tell me how to figure it out myself, or if you could tell me which import I need to have the ShadowRootMode enum, both would be fine!

Thank you for your help!

**Comment by p-ob** [comment]
Short answer: `{...LitElement.shadowRootOptions, mode: 'open', delegatesFocus: true}; }`. Using the literal string value is what would be expected.

I'm unsure how to get the enum ref if you really wanted that, but it wouldn't be importable from 'lit' as that isn't a type defined by the lit package. It's a baked-in enum to the TypeScript library (`lib.dom.d.ts`).

**Comment by aschaepper** [comment]
Hello
Unfortunately that does not work, then I get the errormessage:

"Class static side 'typeof RieterWebcomponents' incorrectly extends base class static side 'typeof LitElement'.
  The types of 'shadowRootOptions.mode' are incompatible between these types.
    Type 'string' is not assignable to type 'ShadowRootMode'.ts(2417)"
   
And as you mentioned, I cant use the ShadowRootMode Enum anyway because its not marked with "export" inside the Typescript library. So I really don't get how one should use this static field and how its ment to be used from the documentation.

**Reply by itsmebasti** [reply, marked as answer]
this is the solution to this, it's rather TS related, I also ran into this yesterday.
`{ ...LitElement.shadowRootOptions, mode: 'open' as const }`
